backend/src/main/java/ai/verta/modeldb/project/FutureProjectDAO.java

```diff
@@ -343,4 +379,358 @@ public FutureProjectDAO(
     return permissionCheck.thenCompose(
         unused -> artifactHandler.getUrlForArtifact(request), executor);
   }
+
+  public InternalFuture<FindProjects.Response> findProjects(FindProjects request) {
```

Comment:
This method has hundreds of lines. It's even hard to read with the indentation! Let's split it up.

Reply:
Splatted findProjects function in multiple functions